### PR TITLE
send unselected row to very bottom when send-to-bottom is clicked

### PR DESCRIPTION
**Pull request by nickpell**
https://manage.dimagi.com/default.asp?277910

### CODE REVIEW

File: corehq/apps/hqwebapp/static/hqwebapp/js/knockout_bindings.ko.js [hunk 1]
```diff
@@ -260,19 +260,24 @@ hqDefine("hqwebapp/js/knockout_bindings.ko", ['jquery', 'knockout', 'jquery-ui/u
...
-                var currentIndex = row[0].attributes['data-order'].value;
-
-                var lastSelectedRowIndex = null;
-                for (var i = 0; i < list().length; i++) {
-                    if (list()[i].selected()) {
-                        lastSelectedRowIndex = i;
+                var currentIndex = parseInt(row[0].attributes['data-order'].value);
+                if (list()[currentIndex].selected()) {
+                    var lastSelectedRowIndex = null;
+                    for (var i = 0; i < list().length; i++) {
```
Comment (proteusvacuum) [inline corehq/apps/hqwebapp/static/hqwebapp/js/knockout_bindings.ko.js]: not a big deal, but if you looped through the list backwards, you could break out of the loop when you find a single one that matches, which means you don't have to check every element.

File: corehq/apps/hqwebapp/static/hqwebapp/js/knockout_bindings.ko.js [hunk 2]
```diff
@@ -260,19 +260,24 @@ hqDefine("hqwebapp/js/knockout_bindings.ko", ['jquery', 'knockout', 'jquery-ui/u
...
+                            lastSelectedRowIndex = i;
+                        }
                     }
-                }
 
-                if (currentIndex < lastSelectedRowIndex) {
+                    if (currentIndex < lastSelectedRowIndex) {
+                        var currentListItem = list.splice(currentIndex, 1)[0];
+                        row.remove();
+                        list.splice(lastSelectedRowIndex, 0, currentListItem);
```
Comment (proteusvacuum) [inline corehq/apps/hqwebapp/static/hqwebapp/js/knockout_bindings.ko.js]: nit: this function is now really long, and has duplicated functionality. Do you think you could break some of this logic out into named functions so it is easier to read? Like this whole if statement could be called `sendToBottom` or something.

**Comment by nickpell** [comment]
@proteusvacuum updated

**Review by proteusvacuum — APPROVED** [review]
thanks!

**Pull request merged**